### ISSUE TITLE
something about the stop gradient

### ISSUE THREAD

**Issue by niujiaopeng** [HiKapok/SSD.TensorFlow#52]
thank you for your great job!
I have carefully read your code, there is only on question:
1.the final mask contains the positive mask and the hard mining negative mask, In your code ,I think we need  to use only the boxes locate in the final mask to calc gradient,  but tf.stop_gradient will stop computing gradient into account. is it just doing the opposite
2.Why flaten_cls_targets use final_mask which contains negative and positive,the flaten_loc_targets only contains the positive
thanks

**Comment by HiKapok** [comment]
@niujiaopeng 1. stop_gradient only stop gradient flow back to masks, since those logical function only outputs True or False which isn't differentiable. 2. for classcification we can use both positive and negtive anchors, while for regression, those negtive anchors has no ground truth with them, so we just use positive anchors
